models/dataset.go

```diff
@@ -0,0 +1,84 @@
+package models
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Datasets represents the model returned from the api datasets
```

Comment:
This is not what we want the model to look like, it is a model that describes the data being pulled out of dp-graph, the dataset API needs to manipulate the structure that dp-graph returns to it, so it can shape the date to meet it's needs